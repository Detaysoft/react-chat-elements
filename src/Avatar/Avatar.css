.rce-avatar-container {
    overflow: hidden;
    display: flex;
    justify-content: center;
    align-items: center;
    overflow: hidden;
}

.rce-avatar-container .rce-avatar {
    width: 100%;
    height: 100%;
}

.rce-avatar-container.flexible .rce-avatar {
    height: auto !important;
    width: 100% !important;
    border-radius: unset !important;
    overflow: unset !important;
}

.rce-avatar-container.default{
    width: 25px;
    height: 25px;
}

.rce-avatar-container.rounded{
    border-radius: 5px;
}

.rce-avatar-container.circle{
    border-radius: 100%;
}

.rce-avatar-container.xsmall{
    width: 30px;
    height: 30px;
}

.rce-avatar-container.small{
    width: 35px;
    height: 35px;
}

.rce-avatar-container.medium{
    width: 40px;
    height: 40px;
}

.rce-avatar-container.large{
    width: 45px;
    height: 45px;
}

.rce-avatar-container.xlarge{
    width: 55px;
    height: 55px;
}
