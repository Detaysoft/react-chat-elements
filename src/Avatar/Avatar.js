import React, { Component } from 'react';
import './Avatar.css';

const classNames = require('classnames');

export class Avatar extends Component {
    render() {
        return (
            <div className={classNames('rce-avatar-container', this.props.type, this.props.size, this.props.className)}>
                <img alt={this.props.alt} src={this.props.src} className={'rce-avatar'} />
                {this.props.sideElement}
            </div>
        );
    }
}

Avatar.defaultProps = {
    type: 'default',
    size: 'default',
    src: '',
    alt: '',
    sideElement: null,
};

export default Avatar;
