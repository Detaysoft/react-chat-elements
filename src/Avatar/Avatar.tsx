import { useEffect } from 'react'
import './Avatar.css'
import classNames from 'classnames'
import { IAvatarProps } from '../type'

const Avatar: React.FC<IAvatarProps> = ({ type = 'default', size = 'default', lazyLoadingImage = undefined, ...props }) => {
  let loadedAvatars: string[] = []
  let loading: boolean = false
  let src = props.src
  let isLazyImage: boolean = false

  useEffect(() => {
    if (lazyLoadingImage) {
      isLazyImage = true

      if (!isLoaded(src)) {
        src = lazyLoadingImage

        if (!loading) {
          requestImage(props.src)
        }
      } else {
        isLazyImage = false
      }
    }
  }, [])

  const isLoaded = (src: string) => {
    return loadedAvatars.indexOf(src) !== -1
  }

  const requestImage = (src: string) => {
    loading = true

    var loaded = () => {
      loadedAvatars.push(src)
      loading = false
    }

    var img: HTMLImageElement = document.createElement('img')
    img.src = src
    img.onload = loaded
    img.onerror = loaded
  }

  const stringToColour = (str: string) => {
    var hash: number = 0
    for (let i: number = 0; i < str.length; i++) {
      hash = str.charCodeAt(i) + ((hash << 5) - hash)
    }
    var colour: string = '#'
    for (let i: number = 0; i < 3; i++) {
      var value: number = (hash >> (i * 8)) & 0xff
      value = (value % 150) + 50
      colour += ('00' + value.toString(16)).substr(-2)
    }
    return colour
  }

  return (
    <>
        {/* @ts-ignore */}
        <div className={classNames('rce-avatar-container', type, size, props.className)}>
            {props.letterItem ? (
                <div className='rce-avatar-letter-background' style={{ backgroundColor: stringToColour(props.letterItem.id) }}>
                    <span className='rce-avatar-letter'>{props.letterItem.letter}</span>
                </div>
            ) : (
                <img
                    alt={props.alt}
                    src={src}
                    onError={props.onError}
                    className={classNames('rce-avatar', { 'rce-avatar-lazy': isLazyImage })}
                />
            )}
            {props.sideElement}
        </div>

        {props.miniImage && (
            <div className='rce-mini-avatar-container'>
                <img
                    src={props.miniImage}
                />
            </div>
        )}
    </>
  )
}
export default Avatar
