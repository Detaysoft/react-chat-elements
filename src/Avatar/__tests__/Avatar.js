import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Avatar from '../Avatar'

describe('Avatar component', () => {
  it('should render without issues', () => {
    const component = shallow(<Avatar src='' />)
    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
