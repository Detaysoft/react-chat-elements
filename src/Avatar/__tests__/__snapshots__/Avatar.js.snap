// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render without issues 1`] = `
<div
  className="rce-avatar-container default default"
>
  <img
    alt=""
    className="rce-avatar"
    onError={[Function]}
    src=""
  />
</div>
`;
